src/test/communitycommons/XPathTest.java

```diff
@@ -0,0 +1,316 @@
+package communitycommons;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import com.mendix.core.CoreException;
+import com.mendix.core.objectmanagement.member.MendixObjectReference;
+import com.mendix.core.objectmanagement.member.MendixObjectReferenceSet;
+import com.mendix.systemwideinterfaces.core.IContext;
+import com.mendix.systemwideinterfaces.core.IMendixIdentifier;
+import com.mendix.systemwideinterfaces.core.IMendixObject;
+import com.mendix.systemwideinterfaces.core.IMendixObjectMember;
+import com.mendix.systemwideinterfaces.core.meta.IMetaObject;
+
+public class XPathTest {
+
+  @Test
+  public void valueToXPathValue() throws Exception {
+    assertEquals("NULL", XPath.valueToXPathValue(null));
+    assertEquals("42", XPath.valueToXPathValue(42));
+    assertEquals("42.42", XPath.valueToXPathValue(42.42));
+    assertEquals("'string'", XPath.valueToXPathValue("string"));
+    assertEquals("true()", XPath.valueToXPathValue(true));
+    assertEquals("false()", XPath.valueToXPathValue(false));
+
+    var d = new Date();
+    assertEquals("" + d.getTime(), XPath.valueToXPathValue(d));
+
+    var id = new TestIdentifier(123);
+    assertEquals("'123'", XPath.valueToXPathValue(id));
+
+    var obj = new TestObject(id);
+    assertEquals("'123'", XPath.valueToXPathValue(obj));
+
+    var proxy = new TestProxy(obj);
+    assertEquals("'123'", XPath.valueToXPathValue(proxy));
+  }
+
+  // XPath
+  @Test
+  public void simple() throws Exception {
+    var xpath = XPath.create(null, "A");
+    assertEquals("//A", xpath.getXPath());
+  }
+
+  // comparisons
+  @Test
+  public void eq() throws Exception {
```

Comment:
We should add tests for the assertions on the number of optional parameters passed.

Reply:
Added some tests